From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Arthur Eubanks <aeubanks@google.com>
Date: Mon, 10 Jul 2023 20:09:14 +0000
Subject: chore: update to newer clang

Build and package win/aarch64 compiler-rt libraries

Specifically profile lib and builtins.

Bug: 1444484, 1453618
Change-Id: Iae0a6c5d8998b4af0281086349f21c15d3e8fd4a
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4663931
Reviewed-by: Hans Wennborg <hans@chromium.org>
Commit-Queue: Arthur Eubanks <aeubanks@google.com>
Cr-Commit-Position: refs/heads/main@{#1168279}

diff --git a/tools/clang/scripts/package.py b/tools/clang/scripts/package.py
index 00858cc30e9d04cb7a4013ead312b8e9db8acca3..d7d01d9e13f2505b50fc2a20fd388d69a354e934 100755
--- a/tools/clang/scripts/package.py
+++ b/tools/clang/scripts/package.py
@@ -457,10 +457,12 @@ def main():
         # Builtins for C/C++.
         'lib/clang/$V/lib/windows/clang_rt.builtins-i386.lib',
         'lib/clang/$V/lib/windows/clang_rt.builtins-x86_64.lib',
+        'lib/clang/$V/lib/windows/clang_rt.builtins-aarch64.lib',
 
         # Profile runtime (used by profiler and code coverage).
         'lib/clang/$V/lib/windows/clang_rt.profile-i386.lib',
         'lib/clang/$V/lib/windows/clang_rt.profile-x86_64.lib',
+        'lib/clang/$V/lib/windows/clang_rt.profile-aarch64.lib',
 
         # UndefinedBehaviorSanitizer C runtime (pure C won't link with *_cxx).
         'lib/clang/$V/lib/windows/clang_rt.ubsan_standalone-x86_64.lib',
diff --git a/tools/clang/scripts/update.py b/tools/clang/scripts/update.py
index e42f8afa37d89c3bf2507e23ad1395ab6ad2c6fb..2288deb727523171438ac0b2f636e3dd62d79756 100755
--- a/tools/clang/scripts/update.py
+++ b/tools/clang/scripts/update.py
@@ -36,7 +36,7 @@ import zlib
 # Reverting problematic clang rolls is safe, though.
 # This is the output of `git describe` and is usable as a commit-ish.
 CLANG_REVISION = 'llvmorg-17-init-16420-g0c545a44'
-CLANG_SUB_REVISION = 1
+CLANG_SUB_REVISION = 2
 
 PACKAGE_VERSION = '%s-%s' % (CLANG_REVISION, CLANG_SUB_REVISION)
 RELEASE_VERSION = '17'
